src/App.js

```diff
@@ -1,155 +1,94 @@
-  const [successMessage, setSuccessMessage] = useState('')
   const { pathname } = useLocation()
-  const history = useHistory()
-
-  useEffect(() => {
-    if (successMessage === 'Success!') {
-      const timer = setTimeout(() => {
-        history.push('/currently-reading')
-        setSuccessMessage('')
-      }, 5000)
-      return () => clearTimeout(timer)
-    }
-  }, [successMessage, history])
 
-  function getBookcoverPreview(previewEvent) {
-    const preview = previewEvent.target.files[0]
-    const reader = new FileReader()
-    reader.onload = event => {
-      setBookcover(event.target.result)
-    }
-    reader.readAsDataURL(preview)
+  function handleSetUsername(name) {
+    setUsername(name)
   }
-
   function handleSetSearchedBook(searchedBook) {
     setSearchedBook(searchedBook)
   }
 
   function handleCreateNewBook(newBookData) {
-    const {
-      title,
-      authors,
-      readingSince,
-      onPage,
-      thumbnail,
-      isbn10,
-      isbn13,
-      year,
-      publisher,
-      description,
-      subtitle,
-      pages,
-    } = newBookData
     const newBook = {
+      ...newBookData,
       id: nanoid(),
       finished: false,
-      readingSince: readingSince,
       finishedSince: '',
-      onPage: onPage,
-      title: title,
-      authors: authors,
-      thumbnail: !thumbnail ? bookcover : thumbnail,
-      isbn10: isbn10,
-      isbn13: isbn13,
-      year: year,
-      publisher: publisher,
-      description: description,
-      subtitle: subtitle,
-      pages: pages,
     }
     const newBooks = [newBook, ...books]
     setBooks(newBooks)
     setLocalStorage(`books${username}`, newBooks)
-    setSuccessMessage('Success!')
     setSearchedBook('')
   }
 
-  return (
-    <AppContainer>
-      <Switch>
-        <Route exact path="/">
-          {username ? (
-            <Redirect to="/currently-reading" />
-          ) : (
-            <HomeScreen history={history} />
-          )}
-        </Route>
-        <Main>
-          <Route exact path={['/currently-reading', '/library']}>
-            {!username ? (
-              <Redirect to="/" />
-            ) : (
-              <BookList books={books} username={username} status={pathname} />
-            )}
-          </Route>
-          <Route exact path="/add-book">
-            {!username ? (
-              <Redirect to="/" />
-            ) : (
-              <StartAddBook
-                history={history}
-                onHandleSetSearchedBook={handleSetSearchedBook}
-              />
-            )}
+  if (!username) {
+    if (pathname === '/') {
+      return (
+        <AppContainer>
+          <HomeScreen onHandleSetUsername={handleSetUsername} />
+        </AppContainer>
+      )
+    } else {
+      return <Redirect to="/" />
+    }
+  } else {
+    return (
+      <AppContainer>
+        <Switch>
+          <Route exact path="/">
+            <HomeScreen onHandleSetUsername={handleSetUsername} />
```

Comment:
I think it's a bit harder to read with the `if...else`. Maybe we can have a look together next week?

Reply:
Sure! :-) Until then I will think about a better solution! 👍